### PR TITLE
Add /whois command

### PR DESCRIPTION
**Pull request by mheguy**
Rename stat get functions to match others
`/whois <player>` -- and if you explicitly don't want to see their inventory you would use `/whois <player> no`. The specific use case I had in mind was when you want to get some info from the server console about a player.

image

### CODE REVIEW

File: features/redmew_commands.lua [hunk 1]
```diff
@@ -213,6 +216,45 @@ local function print_version()
...
 
+--- Prints information about the target player
+local function print_player_info(args, player)
+    local name = args.player
+    local target = game.players[name]
+    if not target then
+        Game.player_print('Target not found')
+    end
+    local index = player.index
+    local m_inventory = player.get_inventory(defines.inventory.player_main)
```
Comment (grilledham) [inline features/redmew_commands.lua]: You seem to be getting the inventory of the player who ran the command not the target?

File: features/redmew_commands.lua [hunk 2]
```diff
@@ -213,6 +216,45 @@ local function print_version()
...
+    m_inventory = m_inventory.get_contents()
+    local q_inventory = player.get_inventory(defines.inventory.player_quickbar)
+    q_inventory = q_inventory.get_contents()
```
Comment (grilledham) [inline features/redmew_commands.lua]: `m_inventory ` and `q_inventory` only seem to be used below `if player and player.admin and not args.inventory` so why not move them there?

File: features/redmew_commands.lua [hunk 3]
```diff
@@ -213,6 +216,45 @@ local function print_version()
...
+        'Distance walked: ' .. PlayerStats.get_walk_distance(index),
+        'Coin earned: ' .. PlayerStats.get_coin_earned(index),
+        'Coin spent: ' .. PlayerStats.get_coin_spent(index),
+        'Deaths: ' .. PlayerStats.get_death_count(index),
+        'Crafted items: ' .. PlayerStats.get_crafted_item(index),
+        'Chat messages: ' .. PlayerStats.get_console_chat(index)
+    }
+    Game.player_print(concat(info_t, ' - '))
+
+    if player and player.admin and not args.inventory then
```
Comment (grilledham) [inline features/redmew_commands.lua]: If the server runs this command wont `player` be nil?

File: features/redmew_commands.lua [hunk 4]
```diff
@@ -294,19 +336,30 @@ Command.add(
...
-        allowed_by_server = true,
+        allowed_by_server = true
     },
     print_version
 )
 
+Command.add(
+    'whois',
+    {
+        description = 'provides information about a given player, admins get additional information',
```
Comment (grilledham) [inline features/redmew_commands.lua]: Adding that you can hide showing the inventory if you add a second argument might be nice.

File: features/redmew_commands.lua [hunk 5]
```diff
@@ -213,6 +216,45 @@ local function print_version()
...
+        'Surface: ' .. player.surface.name,
+        'Tag: ' .. player.tag,
+        'Distance walked: ' .. PlayerStats.get_walk_distance(index),
+        'Coin earned: ' .. PlayerStats.get_coin_earned(index),
+        'Coin spent: ' .. PlayerStats.get_coin_spent(index),
+        'Deaths: ' .. PlayerStats.get_death_count(index),
+        'Crafted items: ' .. PlayerStats.get_crafted_item(index),
+        'Chat messages: ' .. PlayerStats.get_console_chat(index)
+    }
+    Game.player_print(concat(info_t, ' - '))
```
Comment (grilledham) [inline features/redmew_commands.lua]: Just a thought but adding `\n` to the separator might make this easier to read from the server console, and will be treated as `' '` while in game

File: features/redmew_commands.lua (hunk 1 above)
Comment (mheguy) [reply features/redmew_commands.lua]: Not just the inventory... *facepalm*

**Review by linaori — APPROVED** [review]

File: features/player_create.lua [hunk 6]
```diff
@@ -62,6 +62,7 @@ local function player_created(event)
 
     if _DEBUG and player.admin then
         UserGroups.add_regular(player.name)
+        game.print("HEY GRILLED DON'T LAUNCH THIS MAP! DEBUG MODE IS ENABLED!!!")
```
Comment (grilledham) [inline features/player_create.lua:65]: Because I'm so paranoid about this, I've never actually done it.

**Review by grilledham — APPROVED** [review]

File: features/player_create.lua (hunk 6 above)
Comment (mheguy) [reply features/player_create.lua:65]: Now that the warning is in place, you're sure to do it. :D

**Pull request merged**